load_balancers.go

```diff
@@ -149,6 +154,33 @@ func (s StickySessions) String() string {
 	return Stringify(s)
 }
 
+// LBFirewall holds the allow and deny rules for a loadbalancer's firewall
+// currently allow and deny rules support cidrs and ips
+// please use the helper methods (IPSourceFirewall/CIDRSourceFirewall) to format the allow/deny rules
+type LBFirewall struct {
+	Allow []string `json:"allow,omitempty"`
+	Deny  []string `json:"deny,omitempty"`
+}
+
+func (lbf *LBFirewall) deepCopy() *LBFirewall {
+	return &LBFirewall{
+		Allow: append([]string(nil), lbf.Allow...),
```

Comment:
Could also do `lbf.Allow[:]`

Reply:
i don't think `lbf.Allow[:]` would do a full copy; that is, it might allocate a new slice header but it still point to the same underlying array.

according to [this](https://github.com/golang/go/wiki/SliceTricks#copy), the append method is correct for a full deep copy

Reply:
TIL